pkg/build/util/generate.go

```diff
@@ -154,3 +160,19 @@ func SubstituteImageReferences(build *buildapi.Build, oldImage string, newImage
 		}
 	}
 }
+
+// GetBuildPodName returns name of the build pod.
+func GetBuildPodName(build *buildapi.Build) string {
+	return fmt.Sprintf("pod-%s", build.Name)
+}
+
+// GetNextBuildName returns name of the next build and increments BuildConfig's LastVersion.
+func GetNextBuildName(bc *buildapi.BuildConfig) string {
+	bc.LastVersion++
+	return fmt.Sprintf("%s-%d", bc.Name, bc.LastVersion)
+}
+
+// getNextBuildNameFromBuild returns name of the next build with random uuid added at the end
+func getNextBuildNameFromBuild(build *buildapi.Build) string {
+	return fmt.Sprintf("%s-%s", build.Name, uuid.NewUUID().String())
```

Comment:
You won't have retry until you port over to the new framework, so you're dramatically increasing the chance of a collision because the search space is smaller.

----- Original Message -----
> > @@ -154,3 +160,19 @@ func SubstituteImageReferences(build *buildapi.Build,
> > oldImage string, newImage
> >  		}
> >  	}
> >  }
> > +
> > +// GetBuildPodName returns name of the build pod.
> > +func GetBuildPodName(build *buildapi.Build) string {
> > +	return fmt.Sprintf("pod-%s", build.Name)
> > +}
> > +
> > +// GetNextBuildName returns name of the next build and increments
> > BuildConfig's LastVersion.
> > +func GetNextBuildName(bc *buildapi.BuildConfig) string {
> > +	bc.LastVersion++
> > +	return fmt.Sprintf("%s-%d", bc.Name, bc.LastVersion)
> > +}
> > +
> > +// getNextBuildNameFromBuild returns name of the next build with random
> > uuid added at the end
> > +func getNextBuildNameFromBuild(build *buildapi.Build) string {
> > +	return fmt.Sprintf("%s-%s", build.Name, uuid.NewUUID().String())
> 
> I just meant calling the GenerateName function directly with the original
> build.Name as a prefix.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/1275/files#r26507050